src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -127,6 +128,33 @@ public virtual IEdmProperty ResolveProperty(IEdmStructuredType type, string prop
             return type.FindProperty(propertyName);
         }
 
+        /// <summary>
+        /// Resolve term name from model.
+        /// </summary>
+        /// <param name="model">The model to be used.</param>
+        /// <param name="termName">The term name to be resolved.</param>
+        /// <returns>Resolved term.</returns>
+        public virtual IEdmTerm ResolveTerm(IEdmModel model, string termName)
+        {
+            if (EnableCaseInsensitive)
+            {
+                var result = model.SchemaElements.OfType<IEdmTerm>()

```

Comment:
>model.SchemaElements [](start = 29, length = 20)

how about if the input term is in CoreModel?

I am wondering "SchemaElements" includes the Core annotation and capabilities annotation.

Reply:
I don't think capabilities would ever be projected; will have to think about Core (maybe permissions?)

---
In reply to: [166082192](https://github.com/OData/odata.net/pull/1063#discussion_r166082192) [](ancestors = 166082192)

Reply:
Added code to traverse referenced schemas, which would include core, etc.

---
In reply to: [166162587](https://github.com/OData/odata.net/pull/1063#discussion_r166162587) [](ancestors = 166162587,166082192)